### PR TITLE
gh-115077: Make some Argument Clinic error messages more helpful

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-115077

### CODE REVIEW

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -5019,7 +5021,8 @@ def at_classmethod(self) -> None:
 
     def at_critical_section(self, *args: str) -> None:
         if len(args) > 2:
-            fail("Up to 2 critical section variables are supported")
+            fail("Only 2 critical section variables are supported; "
+                 f"{len(args)} was given")
```
Comment (sobolevn) [inline Tools/clinic/clinic.py]:
```suggestion
                 f"{len(args)} were given")
```

Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Perhaps also print them out?

**Comment by erlend-aasland** [comment]
@sobolevn, feel free to contribute to this branch/PR.

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -834,7 +834,9 @@ def output_templates(
                     min_kw_only = i - max_pos
             elif p.is_vararg():
                 if vararg != self.NO_VARARG:
-                    fail("Too many var args")
+                    assert isinstance(vararg, int)
+                    fail("Cannot specify multiple varargs; "
+                         f"'*{parameters[vararg].name}' was already provided as parameter {vararg+1}")
```
Comment (serhiy-storchaka) [inline Tools/clinic/clinic.py]: You refer to the same parameter by the name and the number.
But the conflict is between two varargs `parameters[vararg].name` (number `vararg+1`) and `p.name` (number `i`).

Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Perhaps the number is not needed. I agree it is better if we refer to both parameters by name.

**Comment by erlend-aasland** [comment]
Perhaps 94e082e45be8c354833980402badc9d5eabc900e is overdoing it.

**Comment by erlend-aasland** [comment]
I'll have to start this branch over again; closing.

**Pull request closed without merging**